src/sentry/integrations/slack/utils/errors.py

```diff
@@ -0,0 +1,51 @@
+import logging
+from dataclasses import dataclass
+
+from slack_sdk.errors import SlackApiError, SlackRequestError
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, eq=True)
+class SlackSdkErrorCategory:
+    message: str
+    check_body: bool = False
+
+
+SLACK_SDK_ERROR_CATEGORIES = (
+    RATE_LIMITED := SlackSdkErrorCategory("ratelimited"),
+    CHANNEL_NOT_FOUND := SlackSdkErrorCategory("channel_not_found"),
+    EXPIRED_URL := SlackSdkErrorCategory("Expired url", check_body=True),
+)
+
+_CATEGORIES_BY_MESSAGE = {c.message: c for c in SLACK_SDK_ERROR_CATEGORIES}
+
+
+def unpack_slack_api_error(exc: SlackApiError | SlackRequestError) -> SlackSdkErrorCategory | None:
+    """Retrieve the Slack API error category from an exception object.
+
+    Check three places in priority order:
+    1. the error field of the server response;
+    2. the first line of the message body; and,
+    3. for categories with the `check_body` flag, the rest of the message.
+    """
+
+    if isinstance(exc, SlackApiError):
+        try:
+            error_attr = exc.response["error"]
+            return _CATEGORIES_BY_MESSAGE[error_attr]
+        except KeyError:
+            pass
+
+    dump = str(exc)
+    category = _CATEGORIES_BY_MESSAGE.get(dump.split("\n")[0])
+    if category:
+        return category
+
+    for category in SLACK_SDK_ERROR_CATEGORIES:
+        if category.check_body and category.message in dump:
+            return category
```

Comment:
the "Expired url error" still seems iffy to me. maybe we log here to double check if we can rely on any of the other places instead?

Reply:
Discussed with @cathteng and @iamrajjoshi; resolved that we'll add metrics in a follow-up PR.